### PR TITLE
`@penumbra-zone/wasm`

### PR DESCRIPTION
**Pull request by turbocrime**
work on #737

the wasm package on npm is outdated, and we have consumers trying to use, so i took a shot at this

## keys package
* another pr #958

## wasm package
* removed /src/ path segment
* publishes without build as per discussion
* keys are now an optional dependency

verified working on an example nextjs app with the output of `pnpm pack` which you can view at https://github.com/turbocrime/nextjs-penumbra-wasm-example/

### CODE REVIEW

**Comment by turbocrime** [comment]
@grod220 the 'keys' package should not be versioned with changeset, as it corresponds directly to the penumbra core repo version. but i'm not sure how publishing works when we're not using changeset to manage. could use some attention.

File: packages/wasm/package.json [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
-    "@penumbra-zone/keys": "workspace:*",
     "@penumbra-zone/types": "workspace:*",
     "zod": "^3.22.4"
   },
   "devDependencies": {
     "@penumbra-zone/bech32": "workspace:*",
     "fake-indexeddb": "^5.0.2"
+  },
+  "optionalDependencies": {
+    "@penumbra-zone/keys": "workspace:*"
```
Comment (grod220) [inline packages/wasm/package.json:30]: Interesting, never heard of this category of deps

Comment (grod220) [reply packages/wasm/package.json:30]: Is it required to be a dynamic import like so? https://github.com/turbocrime/nextjs-penumbra-wasm-example/blob/360201378c47cc217a704ceaf8b32bce72e120a9/app/components/ComponentWithWasm.tsx#L6

File: packages/wasm/package.json [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
   "version": "3.0.0",
-  "private": true,
   "license": "MIT",
   "type": "module",
   "scripts": {
     "clean": "rm -rfv dist wasm ; cargo clean --manifest-path ./crate/Cargo.toml",
     "dev": "cargo watch -C ./crate --postpone -- pnpm turbo run compile",
     "compile": "cd crate ; wasm-pack build --no-pack --target bundler --out-name index --out-dir ../wasm",
-    "build": "tsc",
+    "prepack": "touch ./wasm/.npmignore",
```
Comment (grod220) [inline packages/wasm/package.json:10]: What does this do/why is it required?

**Review by grod220 — APPROVED** [review]
Nice job! After this is merged, the bot will update this PR: https://github.com/penumbra-zone/web/pull/940. Then if you merge that, this will publish to npm for the consumer needing this.

Also, could you add documentation to a README.md on how to use this package? Think it would be helpful to give instructions on:
- No keys usage
- Usage w/ keys (and downloading that optional package)
- Possible required webpack/next configs

File: packages/wasm/package.json (hunk 1 above)
Comment (turbocrime) [reply packages/wasm/package.json:30]: this is the way i could get it to work with next. will try more today

Comment (turbocrime) [reply packages/wasm/package.json:30]: note that isn't the keys package import that's a submodule that happens to be named keys

File: packages/wasm/package.json (hunk 2 above)
Comment (turbocrime) [reply packages/wasm/package.json:10]: this prevents ignoring the compiled wasm when packaging by overriding the wildcard gitignore

Comment (turbocrime) [reply packages/wasm/package.json:10]: prepack is called during publishing

**Pull request merged**